internal/test/crds.go

```diff
@@ -0,0 +1,115 @@
+/*
```

Comment:
I named the package `test` as it's very simple and consistent with our `envtest` package (and I couldn't think of a better name right now). I'm not sure if we want to either name the package differently or further divide it into sub-packages.

But let's discuss and see if we can find a better name. (/cc @vincepri)

Some alternatives:
* testing (like Go's testing package)
* fake (frequently used in client-go for fake clients, although we don't intend to implement a fake client here)

Reply:
+1 for testing

Reply:
`testing` is going to conflict with the Go testing package.

How about `testhelp` or, even more specifically `testtypes`?

Reply:
If we go for something like `testtypes`, we can definitely extend it and implement some builders in there (fake, or semi-fake) to generate the objects we need during testing.

I'd also split up the types in multiple files, based on the API group

Reply:
Good point about testing. I'll go with `testtypes` for now.

Reply:
Two comments:
- testtypes could be limiting in the future given that we plan to add not only object builders, but also matchers (e.g condition matcher) or other utils, what about testutil
- assuming that in the future we are going to add also utils for CAPI types, does the file naming requires to get them all in a single file, this will be impractical. I assume we will need ~1 file for each type + eventually something like groupversion_info.go; but we can get there in next iterations...

Reply:
Maybe the best solution is to move to sub-packages (maybe with follow-up PRs). So this could look like:
* test
    * types
    * builders
    * <other package names depending on the content>

In this case it would still be kind of idiomatic to import the packages as testtypes, .. (at least in case of conflicts). But probably easier to reason about as our test packages grow and because it's all internal it's easy to change.

Reply:
I like the idea of having smaller, focused packages. 
I would suggest to use internal/testing as a top level package and move envtest there too

Reply:
As per comment in https://github.com/kubernetes-sigs/cluster-api/pull/5040#issuecomment-891084020 we can eventually iterate later on package struct. For now, let's start moving this out of /controllers